js/foam/browser/u2/StackView.js

```diff
@@ -0,0 +1,396 @@
+      $ {
+        height: 100%;
+        overflow-x: hidden;
+        position: relative;
+        width: 100%;
+      }
+      $-panel {
+        background-color: #fff;
+        display: flex;
+        flex-direction: column;
+        height: 100%;
+        overflow: hidden;
+        position: absolute;
+      }
+      $-edge {
+        background-color: #000;
+        height: 100%;
+        opacity: 0.1;
+        position: absolute;
+        right: 0px;
+        top: 0px;
+        width: 1px;
+        z-index: 100;
+      }
+
+      $-hidden {
+        display: none;
+      }
+      $-no-input {
+        pointer-events: none;
+      }
+    */},
+  ],
+
+  listeners: [
+    {
+      name: 'onLoad',
+      code: function() {
+        // Render and configure each child view that has already been loaded.
+        for (var i = 0; i < this.views_.length; i++) {
+          this.renderChild(i);
+        }
+        this.resize();
+      }
+    },
+    {
+      name: 'onResize',
+      isFramed: true,
+      code: function() {
+        this.resize();
+      }
+    },
+    {
+      name: 'pushView_',
+      isFramed: true,
+      documentation: function() {/*
+        <p>The real implementation used to push new views onto the stack.
+        The index specifies the level of the stack we're operating at.
+
+        Generally this is not called directly. Instead, views use the stack in
+        their contexts to push views. That method is a bound call to this function
+        with the index set appropriately.
+
+        <h4>Hints</h4>
+        <p>The last argument is an object containing hints. Should the switch
+        be transitionless, or backwards? The supported hints are documented here.
+        They take the form of an object with options rather than a confusing
+        series of optional arguments.</p>
+        <dl>
+          <dt>instant: true</dt>
+          <dd>If this StackView is configured to use an animated transition,
+          setting <tt>instant: true</tt> will skip the transition and instantly
+          replace the view.</dd>
+
+          <dt>preview: true</dt>
+          <dd>Previews are optional pushes. This mode might be used to push and
+          render details of the item under the cursor. Show the child view (the
+          preview) if there's room for both the main and child views, but if
+          there isn't room then keep showing the main view.</dd>
+        </dl>
+      */},
+      // TODO(braden): Actually implement the preview hint. Instant is a no-op
+      // until we actually have animated transitions to skip.
+      code: function(index, view, hints) {
+        if (!view) return;
+        if (!hints) hints = {};
+
+        // Pushing a new view with index i means we drop everything in the list
+        // that's greater than i, and then add the new view back. There will be
+        // i + 1 views in the stack after a pushView(i, v);
+        this.destroyChildViews_(index + 1);
+
+        var substack = {
```

Comment:
It would be nice to model substack instead of using __proto__.  This approach will be useful when we want to model the Toolbar object that goes in the context as well.